sdks/java/core/src/main/java/org/apache/beam/sdk/util/BitSetCoder.java

```diff
@@ -58,4 +59,10 @@ public void verifyDeterministic() throws NonDeterministicException {
         "BitSetCoder requires its byteArrayCoder to be deterministic.",
         byteArrayCoder);
   }
+
+  private void readObject(ObjectInputStream in)
+      throws IOException, ClassNotFoundException {
+    in.defaultReadObject();
+    this.byteArrayCoder = ByteArrayCoder.of();
+  }
```

Comment:
hmm.

Probably a better solution is to make the `byteArrayCoder` `private static final`, and of course then also non-`transient`? If it's static (which it should be) it wouldn't get serialized anyway. Simpler and then we don't need `readObject`.